### PR TITLE
Add PHPMD

### LINKED ISSUE #106: Setup PHPMD in repo
### Is your enhancement related to a problem? Please describe.

[PHPMD](https://phpmd.org/) is tool similar to Java's PMD.

PHPMD is a useful tool, for find overly complex code, unused code, code path that will never be reached and other issues. 

PHPMD, like phpstan can be introduced in the CL pipeline and can be seen as mark of quality if CI tests are passed. 

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by swissspidy**
### Description of the Change

PHPMD is a static analysis tool that helps identify overly complex code paths and similar things.

This PR adds an initial config including some fixes for low hanging fruit for some of the reported errors, as well as error suppression for the reported NPath errors, which deserve their own dedicated issues.

NPath describes the number of possible code paths. The lower the better, as it makes testing and maintenance easier due to the reduced complexity. Often times this can be achieved by extracting parts of the logic into separate methods.

Closes #106

### Changelog Entry

> Added - New feature
> Changed - Existing functionality
> Deprecated - Soon-to-be removed feature
> Removed - Feature
> Fixed - Bug fix
> Security - Vulnerability

### Credits

Props @username, @username2, ...

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@swissspidy This PR looks good. I think we could merge it.

**Comment by swissspidy** [comment]
Well not with failing checks :-) Let me address those first.

File: includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php [hunk 1]
```diff
@@ -88,7 +89,7 @@ final public function run( Check_Result $result ) {
 		// Run PHPCS.
 		try {
 			ob_start();
-			$runner = new \PHP_CodeSniffer\Runner(); // phpcs:ignore SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName
+			$runner = new Runner();
```
Comment (spacedmonkey) [inline includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php:92]: Should we also change `if ( ! class_exists( '\PHP_CodeSniffer\Runner' ) ) {` above?

**Review by spacedmonkey — APPROVED** [review]
One nicpic, otherwise, this looks good.

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks @swissspidy, Great work!

**Pull request merged**